nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -526,13 +531,153 @@ def conformer_stream_step(
             if all_hyp_or_transcribed_texts is None:
                 all_hyp_or_transcribed_texts = best_hyp
 
-        return (
+        result = [
             greedy_predictions,
             all_hyp_or_transcribed_texts,
             cache_last_channel_next,
             cache_last_time_next,
             best_hyp,
-        )
+        ]
+        if return_log_probs:
+            result.append(log_probs)
+            result.append(encoded_len)
+
+        return tuple(result)
+
+    @torch.no_grad()
+    def transcribe_simulate_cache_aware_streaming(
```

Comment:
If you just want the final transcriptions, you don't need this method and no need to simulate streaming. If you call the regular transcribe method, it gives you the exact same result as this method. When you have the whole audio, the result of offline and streaming is exactly the same for cache-aware models.

Reply:
I need the log_probs. If I just use the regular transcribe() method, it will cause GPU OOM issue for long audios.